integrations/java/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/exception/WebClientResponseWithMessageException.java

```diff
@@ -8,21 +8,37 @@
  *
  * <p>In this class we override the {@link #getMessage()} behavior, to append server response if it
  * exists. It doesn't change status code (ex: forbidden) or status int (ex. 403)
+ *
+ * <p>An exception thrown in Openhouse clients to indicate an error in a tables API request, acting
+ * as a wrapper around WebClientResponseException.
  */
-public class WebClientResponseWithMessageException extends WebClientResponseException {
-  public WebClientResponseWithMessageException(WebClientResponseException e) {
-    super(
+public class WebClientResponseWithMessageException extends WebClientWithMessageException {
+  private final WebClientResponseException responseException;
+
+  /**
+   * We want the response exception to go through the legacy constructor in {@link
+   * WebClientResponseException} because the HttpRequest value is explicitly set to null in order to
+   * have the message rewritten to not expose unnecessary data.
+   */
+  public WebClientResponseException createWebClientResponseException(WebClientResponseException e) {
```

Comment:
`super(exception)` calling the constructor of `WebClientWithMessageException` does not recreate the same behaviour as `super` calling `WebClientResponseException` because `WebClientResponseException` has [initMessage](https://github.com/spring-projects/spring-framework/blob/main/spring-webflux/src/main/java/org/springframework/web/reactive/function/client/WebClientResponseException.java#L100) which will add information into the error message (ex: `401 Unauthorized from PATCH http://localhost:56935/v1/databases/dgrant/tables/t1/aclPolicies` in the tests) if the HttpRequest is not explicitly set as `null` like it was before.